tests/components/select/test_init.py

```diff
@@ -26,3 +51,88 @@ async def test_select(hass: HomeAssistant) -> None:
     select._attr_current_option = "option_four"
     assert select.current_option == "option_four"
     assert select.state is None
+
+    select.hass = hass
+
+    with pytest.raises(NotImplementedError):
+        await select.async_select_option("option_one")
+
+    select.select_option = MagicMock()
+    await select.async_select_option("option_one")
+
+    assert select.select_option.called
+    assert select.select_option.call_args[0][0] == "option_one"
+
+    assert select.capability_attributes[ATTR_OPTIONS] == [
+        "option_one",
+        "option_two",
+        "option_three",
+    ]
+
+
+async def test_custom_integration_and_validation(
+    hass, entity_reg, enable_custom_integrations
+):
+    """Test we can only select valid options."""
+    platform = getattr(hass.components, f"test.{DOMAIN}")
+    platform.init()
+
+    reg_entry_1 = entity_reg.async_get_or_create(
```

Comment:
Why do we need to create entity registry entries? It should be enough to set up the custom integration platform.